### PR TITLE
Lending library. Fixes #98

### LINKED ISSUE #98: /books
We have a bunch of books available for loan, but they're currently hosted statically on my website. It'd be great if we had a simple lending library.

### PR DESCRIPTION
**Pull request by tlynam**
This is based on the Rails-4 upgrade branch

### CODE REVIEW

File: app/controllers/books_controller.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      end
+    end
+  end
+
+  private
+
+  def book_params
+    params.require(:book).permit(:title, :author, :name, :email, :comment, :username)
+  end
+end
```
Comment (phiggins) [inline app/controllers/books_controller.rb]: The whitespace is off in several files. It looks like you might be mixing tabs and spaces?

File: app/views/books/_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    <%= f.text_field :author %>
+  </div>
+  <div class="field">
+    <%= f.label :name, 'Your name' %>
+    <%= f.text_field :name %>
+  </div>
+  <div class="field">
+    <%= f.label :email, 'Your email' %>
+    <%= f.text_field :email %>
+  </div>
```
Comment (phiggins) [inline app/views/books/_form.html.erb]: Should we require a user to have an account and be logged in to submit a book?

File: test/controllers/books_controller_test.rb [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                             :title => "Winnie The Pooh",
+                             :author => "A. A. Milne",
+                             :name => "Tigger",
+                             :email => "tigger@gmail.com",
+                             :comment => "Favorite book"
+                            }
+    end
+
+    assert_redirected_to books_path
+    get :index
```
Comment (phiggins) [inline test/controllers/books_controller_test.rb]: I think there is a `follow_redirect!` method or something that will let you get around manually going to the index.

**Comment by phiggins** [comment]
This looks mostly good to me.

@zenspider You requested this one, is there any functionality missing?

File: app/controllers/books_controller.rb (hunk 1 above)
Comment (tlynam) [inline app/controllers/books_controller.rb]: Thanks, fixed editor :)

File: test/controllers/books_controller_test.rb (hunk 3 above)
Comment (tlynam) [inline test/controllers/books_controller_test.rb]: Is that for integration tests?  Should we have integration tests?

http://api.rubyonrails.org/classes/ActionDispatch/Integration/RequestHelpers.html

File: db/seeds.rb [hunk 4]
```diff
@@ -444,25 +444,25 @@
...
-  { :id => 35, :name => "Ivan Stork", :email => "ivan@sustainablemarketing.com", :bio => "", :website => "", :ruby_gems_id => "ivanoats", :featured => false, :twitter => "ivanoats", :github => "ivanoats", :image_url => "http://a0.twimg.com/profile_images/2987698956/c3ffa2792cb597647f2ca3db7f461c4d_bigger.jpeg", :verified => true }
+  { :id => 16, :name => "Mike Mondragon", :email => "mikemondragon@gmail.com", :bio => "Ruby makes me happy.", :website => "http://plasti.cx/", :ruby_gems_id => "mikemondragon", :featured => false, :twitter => "monde", :github => "monde", :image_url => "http://a0.twimg.com/profile_images/52075135/wallace2_bigger.png", :verified => true, :password => 1234 },
```
Comment (tlynam) [inline db/seeds.rb]: Hi @phiggins, this should be its own pr right?

File: app/assets/stylesheets/seattlerb.css [hunk 5]
```diff
@@ -7399,3 +7422,21 @@ abbr.amp {
...
+  list-style: none;
+  margin-top: 5px;
+}
+
+.book_list {
+  list-style: none;
+  margin-top: 5px;
+}
+
+.book_list li {
```
Comment (tlynam) [inline app/assets/stylesheets/seattlerb.css]: Hi @phiggins, is this how we specify we want the li elements within this book_list class to have these attributes?

File: app/controllers/books_controller.rb [hunk 6]
```diff
@@ -0,0 +1,56 @@
+class BooksController < ApplicationController
+  skip_before_filter :verify_authenticity_token, only: [:destroy]
```
Comment (tlynam) [inline app/controllers/books_controller.rb]: Hi @phiggins, sorry for all the questions.  I was trying to verify the book belongs to the current_member in order to delete it.  For some reason, skipping this was the only way for current_member to be available to the destroy action.

File: app/controllers/books_controller.rb [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+  before_action :verify_member, only: [:edit, :update, :destroy]
```
Comment (tlynam) [inline app/controllers/books_controller.rb]: Is this a good way to verify member actions?

File: config/initializers/devise.rb [hunk 8]
```diff
@@ -23,4 +23,7 @@
   else
     ENV.fetch("SECRET_KEY", "dc2d7e3da039a322d8028134c256f97eb18e8c1f24829be0238a11e95ce82f771a94124997d9d11341a3bd62717e7a4239ae0e7f0e014f2ed5e95c586fadf723")
   end
+
+  # The default HTTP method used to sign out a resource. Default is :delete.
+  config.sign_out_via = :get
```
Comment (tlynam) [inline config/initializers/devise.rb]: I had a routing error with the delete method and not sure why

**Comment by tlynam** [comment]
Hi @phiggins, this failed on CI but passes locally.  Any idea?

```
1) Failure: MemberTest#test_invalid_github [/home/travis/build/seattlerb/seattlerb.org/test/models/member_test.rb:57]:
Failed refutation, no message given
```

**Comment by zenspider** [comment]
> On Oct 4, 2015, at 13:15, Todd Lynam <notifications@github.com> wrote:
> 
> Hi @phiggins, this failed on CI but passes locally. Any idea?
> 
> 1) Failure: MemberTest#test_invalid_github [/home/travis/build/seattlerb/seattlerb.org/test/models/member_test.rb:57]:
> Failed refutation, no message given

Shitty assertion. Improve it for more details.

**Comment by tlynam** [comment]
Thanks @zenspider, do you mean it should assert that there's an invalid user error?  Would it be easier to use the github gem [octokit](https://github.com/octokit/octokit.rb)?

**Comment by zenspider** [comment]
`Failed refutation, no message given` means you're using a bare assert, which is why the message is generic and bad. Either write a custom one, or look at `assert_predicate`.

**Comment by zenspider** [comment]
@tlynam where does this stand?

**Comment by tlynam** [comment]
@zenspider after rebasing the tests are now green

**Comment by zenspider** [comment]
Awesome sir. Thank you. I'm going to try to get through all of these in the next few. 

Have to fix deployment first. Ugh.
